src/main/java/edu/kis/powp/jobs2d/command/ImmutableComplexCommand.java

```diff
@@ -0,0 +1,28 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class ImmutableComplexCommand implements ICompoundCommand {
+    private List<DriverCommand> driverCommandList;
+
+    public ImmutableComplexCommand(List<DriverCommand> driverCommandList) {
+        this.driverCommandList = Collections.unmodifiableList(driverCommandList);
+    }
```

Comment:
Note that there might be a problem if you will get a mutable complex command as an element of a driverCommandList since it still can be modified from outside.

Reply:
Update in #106